pinot-core/src/main/java/org/apache/pinot/core/transport/AsyncQueryResponse.java

```diff
@@ -152,12 +153,6 @@ void receiveDataTable(ServerRoutingInstance serverRoutingInstance, DataTable dat
     ServerResponse response = _responseMap.get(serverRoutingInstance);
     response.receiveDataTable(dataTable, responseSize, deserializationTimeMs);
 
-    // Record query completion stats immediately after receiving the response from the server instead of waiting
```

Comment:
IIUC if we remove this discount upon each `receiveDataTable` and rely only on the one in `getFinalResponses`, it means the performance of all fan out servers are determined by the slowest one among them, do you think it would case inaccuracy where we over estimate the load on some servers whenever a time out happens?

Reply:
Good observation.

I see this resulting in more time taken to warm up/ramp up - that's the reason we had this piece of code earlier.  With this approach, we'll be more conservative to not overload servers (because we assume that every server has not responded till the last server responds).  

Achieving both will be a hairier change - considering the interaction between netty/jetty. We can revisit this logic depending on the behavior we see in our environment.